src/ploneorg/addonlisting/cli.py

```diff
@@ -1,28 +1,106 @@
 # -*- coding: utf-8 -*-
 
-# http://pastebin.com/3u1gNCD3
+from ploneorg.addonlisting.utils import update_addon
+from ploneorg.addonlisting.utils import update_addon_list
+from ploneorg.addonlisting.utils import update_addons
 
-# import argparse
+import argparse
 import logging
 
 
 logging.basicConfig()
 log = logging.getLogger('ploneorg.addonlisting-cli')
 
 
-def update_addon_listing():
-    # Parse the command line arguments
-    # argparser = argparse.ArgumentParser()
-    pass
+def cli_update_addon_listing():
+    parser = argparse.ArgumentParser(
+        description="updates the listing with any new addons"
+    )
+    parser.add_argument(
+        "context",
+        help="Path to AddOnFolder relative to Zope root.\
+              Ex:/Plone/<AddOnFolder name>"
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        default=False,
+        help="print more verbose output",
+        action="store_true"
+    )
+    parser.add_argument(
+        "--limit",
+        type=int,
+        default=0,
+        help="limit the number of new addons fetched from PyPI"
+    )
+    parser.add_argument(
+        "-s",
+        "--site-id",
+        dest='site_id',
+        default='Plone',
+        help="ID of the Plone site to fetch the content objects from"
+    )
+    args = parser.parse_args()
+    '''
+    current = app  # noqa
+    for part in site_id.split('/'):
+        current = current[part]
+    portal = current
+    setSite(portal)
+    print portal
 
+    '''
+    # obtain the portal root object somehow and
+    # store in a local variable "portal"
 
-def update_addons():
-    # Parse the command line arguments
-    # argparser = argparse.ArgumentParser()
-    pass
+    context = portal.restrictedTraverse(args.context)
```

Comment:
god catch, traversing is the right direction, but please look if that really work as it might be a Plone only method that did not work on the Zope context.

There you might need to get the Plone Portal Object separately.